Corrected some spelling/wording in YARP node (#35127)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/servers/yarp/header-guidelines.md

---
uid: fundamentals/servers/yarp/header-guidelines

> title: YARP HTTP Header Guidelines

Reviewer: ```suggestion
title: YARP HTTP header guidelines
```

> description: YARP HTTP Header Guidelines

Reviewer: ```suggestion
description: Learn about YARP HTTP header guidelines.
```

ai-usage: ai-assisted
---

> # YARP HTTP Header Guidelines

Reviewer: ```suggestion
# YARP HTTP header guidelines
```

> Headers are a very important part of processing HTTP requests and each have their own semantics and considerations. Most headers are proxied by default, though some used to control how the request is delivered are automatically adjusted or removed by the proxy. The connections between the client and the proxy, and the proxy and the destination, are independent. Therefore, headers that affect the connection and transport need to be filtered. Many headers contain information like domain names, paths, or other details that may be affected when a reverse proxy is included in the application architecture. Below is a collection of guidelines about how specific headers might be impacted and what to do about them.

Reviewer: I see why you added the commas, but the second object seems to take on a non-essential (less important) nature when set off by commas. Perhaps, it would be best to just restate the proposition in this case.

```suggestion
Headers are a very important part of processing HTTP requests and each have their own semantics and considerations. Most headers are proxied by default, though some used to control how the request is delivered are automatically adjusted or removed by the proxy. The connections between the client and the proxy and between the proxy and the destination are independent. Therefore, headers that affect the connection and transport must be filtered. Many headers contain information like domain names, paths, or other details that may be affected when a reverse proxy is included in the application architecture. The following is a collection of guidelines about how specific headers might be impacted and what to do about them.
```

## YARP header filtering

YARP automatically removes request and response headers that could impact its ability to forward a request correctly, or that may be used maliciously to bypass features of the proxy. A complete list can be found [here](https://github.com/microsoft/reverse-proxy/blob/main/src/ReverseProxy/Forwarder/RequestUtilities.cs#L71), with some highlights described below.

> ### Connection, KeepAlive, Close

Reviewer: ```suggestion
### `Connection`, `KeepAlive`, `Close`
```

These headers control how the TCP connection is managed and are removed to prevent impacting the connection on the other side of the proxy.

> ### Transfer-Encoding

Reviewer: ```suggestion
### `Transfer-Encoding`
```

### Distributed tracing headers

These headers include TraceParent, Request-Id, TraceState, Baggage, Correlation-Context.

> They are automatically removed based on `DistributedContextPropagator.Fields`, allowing the forwarding HttpClient to replace them with updated values.

Reviewer: FYI ... The style manual calls for use of contractions. We usually only avoid them when giving weight to the word "not," for example, "Do ***not*** put your client secret in a public configuration file."

Also, if you use the [XREF Gen tool](https://dotnet.github.io/blazor-samples/), you can get the API cross-links quickly.

```suggestion
They're automatically removed based on <xref:System.Diagnostics.DistributedContextPropagator.Fields%2A?displayProperty=nameWithType>, allowing the forwarding <xref:System.Net.Http.HttpClient> to replace them with updated values.
```

You can opt out of modifying these headers by setting `SocketsHttpHandler.ActivityHeadersPropagator` to `null`:

> ```C#

Reviewer: Code languages should also adopt our standards ... "csharp" in this case.

```suggestion
```csharp
```

> ### Host

Reviewer: ```suggestion
### `Host`
```

> The Host header indicates which site on the server the request is intended for. This header is removed by default since the host name used publicly by the proxy is likely to differ from the one used by the service behind the proxy. This can be configured using the [RequestHeaderOriginalHost](xref:fundamentals/servers/yarp/transforms#requestheaderoriginalhost) transform.

Reviewer: ```suggestion
The Host header indicates which site on the server the request is intended for. This header is removed by default since the host name used publicly by the proxy is likely to differ from the one used by the service behind the proxy. This can be configured using the [`RequestHeaderOriginalHost`](xref:fundamentals/servers/yarp/transforms#requestheaderoriginalhost) transform.
```

> ### X-Forwarded-*, Forwarded

Reviewer: ```suggestion
### `X-Forwarded-*`, `Forwarded`
```

> ### Set-Cookie

Reviewer: ```suggestion
### `Set-Cookie`
```

> This response header may contain fields that constrain the path, domain, scheme, etc. where the cookie should be used. Using a reverse proxy may change the effective domain, path, or scheme of a site from the public view. While it would be possible to [rewrite](https://github.com/microsoft/reverse-proxy/issues/1109) response cookies using custom transforms, it's recommended instead to use the Forwarded headers described above to flow the correct values to the destination app so it can generate the correct set-cookie headers.

Reviewer: Latin abbreviations aren't allowed.

```suggestion
This response header may contain fields that constrain aspects of the URL, such as the scheme, domain, or path, where the cookie should be used. Using a reverse proxy may change the effective scheme, domain, or path of a site from the public view. While it would be possible to [rewrite response cookies using custom transforms](https://github.com/microsoft/reverse-proxy/issues/1109), we recommended instead to use the `Forwarded` headers described earlier to flow the correct values to the destination app so it can generate the correct `set-cookie` headers.
```

> ### Location

Reviewer: ```suggestion
### `Location`
```
